api/openapi-spec.yaml

```diff
@@ -0,0 +1,352 @@
+                    type: integer
+                    example: 1
+                  items:
+                    type: array
+                    items:
+                      $ref: "#/components/schemas/EiffelEvent"
+        401:
+          description: Unauthorized
+          content: {}
+        403:
+          description: Forbidden
+          content: {}
+        404:
+          description: The requested events are not found
+          content: {}
+        500:
+          description: Internal server issue
+          content: {}
+  /events/{id}:
+    get:
+      tags:
+      - API
+      summary: To get single event information
+      operationId: getEventUsingGET
+      parameters:
+      - name: id
+        in: path
+        description: "Id of the event."
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Successfully retrieved the Event
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/EiffelEvent"
+        400:
+          description: Wrong type given for parameter
+          content: {}
+        401:
+          description: Unauthorized
+          content: {}
+        403:
+          description: Forbidden
+          content: {}
+        404:
+          description: The requested event is not found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  errorMsg:
+                    type: string
+                    example: "The requested event is not found"
+        500:
+          description: Internal server issue
+          content: {}
+  /search/{id}:
+    post:
+      tags:
+      - API
+      summary: To get upstream/downstream events for an event based on the searchParameters
+        passed
+      operationId: searchUsingPOST
+      parameters:
+      - name: id
+        in: path
+        description: "Id of the event."
+        required: true
+        schema:
+          type: string
+      - name: limit
+        in: query
+        description: >
+          Determines the maximum amount of events to be fetched.
+          Use `-1` for unlimited.
+        schema:
+          type: integer
+          format: int32
+      - name: levels
+        in: query
+        description: >
+          Determines the maximum amount of levels to search.
+          Use `-1` for unlimited
+        schema:
+          type: integer
+          format: int32
+      requestBody:
+        description: |
+          Select what types of links you want upstream/downstream search to follow.
+
+          Examples:
+
+            * Select `CAUSE` if you only want the search to follow `CAUSE` links
+              disregarding other links.
+            * Select `CONTEXT` and `ACTIVITY_EXECUTION` if you want to follow both `CONTEXT`
+              and `ACTIVITY_EXECUTION` links.
+
+          Link Types:
```

Comment:
We should probably add the PRECURSOR link type here

Reply:
Yes, and as a follow up question to that. Are these link types just examples, or should ER bail out if a non-listed link type is given? If they are just examples, then the documentation should clarify that. If they are an absolute list of event types then it should be documented what happens if a non-supported link type is given. I would guess that 403 is thrown if a non-supported link type is given. If that is correct, please expand the description of 403 below to state that.

Reply:
Another wrench I want to throw in here is the 'ALL' type that exists in this array. By having that type we increase the complexity of ERs (if we want to bail out if a non-listen link type is given as @e-backmark-ericsson  said above), since we need to track each link type in the ER and update the ER any time a new link type gets created.

Reply:
* Yes we should add the PRECURSOR link here
* Regarding the list: as I stated in Magnus comment I would like to keep it here (to ease for the user) until we have such a list somewhere else. I will add a 400 error and state that the error is sent when giving a link type that doesn't exist.
* The `ALL` keyword is intended as a wildcard traversing all types of links. To be compliant with our own implementation we would need it. @t-persson any good ideas for a wildcard?

Reply:
I have no problems with the name `ALL`. My problem is with the fact that the ER has to keep track of all link types and has to be updated with every link type addition in the protocol.

Reply:
I have added the new links in ....

Was your concern on the knowing the link connected to the example or to the enum (futher down)?

Reply:
I agree with @t-persson 's comment here that it could be a bad idea to have this minimal api specification so tightly connected to the protocol. Whenever a new link type is added to the protocol we need to add it here as well. It would be better if the ER was independent of Eiffel protocol releases (as much as possible).

Reply:
Fixed in [c809b68](https://github.com/eiffel-community/eiffel-event-repository/pull/12/commits/c809b68f276808320547c89a2010d7e3fc3d5c98)